e2e/simple_klog/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"flag"
+	"os"
+
+	"k8s.io/klog"
+)
+
+var (
+	infoLogLine    = mustBeInEnv("KLOG_INFO_LOG")
+	warningLogLine = mustBeInEnv("KLOG_WARNING_LOG")
+	errorLogLine   = mustBeInEnv("KLOG_ERROR_LOG")
+	fatalLogLine   = mustBeInEnv("KLOG_FATAL_LOG")
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+	klog.Info(infoLogLine)
+	klog.Warning(warningLogLine)
+	klog.Error(errorLogLine)
+	klog.Flush()
+	klog.Fatal(fatalLogLine)
+}
+
+func mustBeInEnv(name string) string {
+	val, ok := os.LookupEnv(name)
+	if !ok {
+		panic(name + " could not be found in environment")
```

Comment:
how about calling all the `mustBeInEnv()` in `main()` and doing a os.Exit(1) on error instead of forcing a panic?

Reply:
I don't care too much. This thing is just there to stand in for an executable that uses `klog`, call it with some `klog` flag, do some `klog` calls and have its `stderr` and such inspected.

I don't mind to implement your proposed changes, but I don't see any value either.

However, probably moving that into a `internal` package and add some comments about the expected use of it might make it clear that this is not a thing you should use or probably touch.

Reply:
forced panics are usually less preferred, but it doesn't matter to much in this case.
will defer to the approvers.